tests/test_prompt.py

```diff
@@ -0,0 +1,178 @@
+"""Tests the xonsh Prompt Formatter."""
+from __future__ import unicode_literals, print_function
+
+import os
+import sys
+import time
+import socket
+from datetime import datetime, timedelta
+from unittest import TestCase, skip, mock
+
+sys.path.insert(0, os.path.abspath('..'))  # FIXME
+
+import nose
+from nose.tools import eq_, ok_
+
+from .tools import mock_xonsh_env
+from xonsh.tools import TERM_COLORS
+
+from xonsh.environ import PromptFormatter, DefaultPromptFormatter
+
+
+class TestPromptFormatter(TestCase):
```

Comment:
nose doesn't have a complete suite of test tools.  It only provides ok_() and eq_().  For other tests you either end up without good test messages, haing to rite out the test messages manually, or reinventing the wheel by reimplementing the unittest tests yourself.